playback/src/mixer/alsamixer.rs

```diff
@@ -1,218 +1,148 @@
-use super::AudioFilter;
+use super::{MappedCtrl, VolumeCtrl};
 use super::{Mixer, MixerConfig};
-use std::error::Error;
 
-const SND_CTL_TLV_DB_GAIN_MUTE: i64 = -9999999;
+use alsa::ctl::{ElemId, ElemIface};
+use alsa::mixer::{MilliBel, SelemChannelId, SelemId};
+use alsa::Ctl;
+
+use std::ffi::CString;
 
 #[derive(Clone)]
-struct AlsaMixerVolumeParams {
+pub struct AlsaMixer {
+    config: MixerConfig,
     min: i64,
     max: i64,
-    range: f64,
-    min_db: alsa::mixer::MilliBel,
-    max_db: alsa::mixer::MilliBel,
     has_switch: bool,
 }
 
-#[derive(Clone)]
-pub struct AlsaMixer {
-    config: MixerConfig,
-    params: AlsaMixerVolumeParams,
+// None of these are Send and cannot be stored in a Mixer struct,
+// so resort to using a macro for DRYness.
+macro_rules! get_simple_element {
+    ($name: ident, $config: expr) => {
+        let mixer = alsa::mixer::Mixer::new(&$config.card, false).expect("Unable to open mixer");
+        let simple_element_id = SelemId::new(&$config.control, $config.index);
+        let $name = mixer
+            .find_selem(&simple_element_id)
+            .expect("Unable to find mixer control");
+    };
 }
```

Comment:
```suggestion
macro_rules! get_simple_element {
    ($config: expr) => {{
        let mixer = alsa::mixer::Mixer::new(&$config.card, false).expect("Unable to open mixer");
        let simple_element_id = SelemId::new(&$config.control, $config.index);
        mixer
            .find_selem(&simple_element_id)
            .expect("Unable to find mixer control")
    }};
}
```

and use 
```rust
let simple_element = get_simple_element!(config);
```

Once again I'm wondering why a simple function does not work here.

Reply:
Didn't think of that.

I still would prefer something like `let mixer = get_mixer(config); let simple_element = get_simple_element(mixer, config);`, what do you think?

Reply:
It's three lines actually, first opening the mixer, then the element ID and finally the element itself.

For reasons of maintainability and consistency I just don't like copying the exact three lines to three functions `new`, `volume` and `set_volume`. Hence the DSL to DRY this up.

Reply:
I might be wrong, but the element ID line could be part of a ´get_simple_element` function. So it's two lines, two very basic lines btw.

Macros are sometimes a bit dishonest. For example, this macro doesn't tell you that it creates a `Mixer` object on the stack of the "caller". It doesn't tell you that it assigns to the passed variable name. It tells you "look, I'm something like a function" although it behaves very differently and is just fancy text replacing. 

Therefore, I think it's sometimes better just to write 2 or 3 lines multiple times if it avoids obfuscating the meaning.

Reply:
Removed macro in 117a2c5.

Reply:
FWIW, when initially writing this, I ran into the similar issue, so check over in https://github.com/diwic/alsa-rs/issues/36 for some suggestions. I did explore the closure option, but got sidetracked and went for the convoluted, but easy approach in the end..